pymc_experimental/distributions/discrete.py

```diff
@@ -171,3 +171,89 @@ def logp(value, mu, lam):
             (-mu / 4) <= lam,
             msg="0 < mu, max(-1, -mu/4)) <= lam <= 1",
         )
+
+
+class BetaNegativeBinomialRV(RandomVariable):
```

Comment:
I suggest you implement this as a CustomDist under the hood, with a dist and logp kwargs.

See https://github.com/pymc-devs/pymc-experimental/pull/239/commits/cbc61fdbce3c6c12127a38393487858ab9fcf9d0

Which only needed to implement the dist method (you also need the logp as you wrote here)

That way the random method will work in other backends automatically and you don't need to bother implementing the moment method as we will have a default after https://github.com/pymc-devs/pymc/pull/6873

Reply:
I migrated to have a similar solution to Chi. Let me know if this is what you had in mind